### PR TITLE
chore: pin dependencies versions

### PR DESCRIPTION
**Pull request by ffenix113**
This will prevent from unexpected issues
with new versions of gems.

We also should think about dropping support for Ruby 2.5.0, as its EOL was over a year ago.

### CODE REVIEW

**Comment by ferhatelmas** [comment]
We didn't lock them so far because we had thought it would be too restrictive for a library but we could.

Regarding 2.5, let's do it.

**Comment by ffenix113** [comment]
> We didn't lock them so far because we had thought it would be too restrictive for a library but we could.

The issue is that some packages(like Faraday) in latest versions require Ruby 2.6 or later, while we support 2.5(at least based on the gemspec, so anyway we can't rely on latest versions of all packages unfortunately.

> Regarding 2.5, let's do it.

I would prefer this to be major bump, but we also don't have much changes since last major bump. What do you think?

**Comment by ferhatelmas** [comment]
> I would prefer this to be major bump in gem version, but we also don't have much changes since last major bump. What do you think?

I think it's fine if we want to follow semver realistically.

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by ferhatelmas** [comment]
@ffenix113 we can release this and then bump ruby version separately and release major too

**Comment by ffenix113** [comment]
> @ffenix113 we can release this and then bump ruby version separately and release major too

Sure, this sounds good.

**Comment by RobinDaugherty** [comment]
If this PR was intended to fix #108, this change should have been to limit the specific dependency to the versions before the broken version (faraday "< 2.5.0" or faraday-net_http "< 3.0.0").

There may be specific versions of some faraday dependencies that are broken, but it's not "all versions of faraday except 1.10.0". Pinning dependencies to a specific version is not a good idea. I don't want to understate that—it's actually a disastrous idea...

Most projects have other dependencies which require faraday, or directly require faraday themselves. Many of these projects might already have a newer version `faraday` or `faraday-multipart` or `faraday-net_http_persistent` than what this PR pins them to.

In projects including a gem, Bundler looks for combinations of dependencies (including transitive dependencies) that meet all restrictions given. It also has a bias to upgrade dependency versions.

Since the releases of the gem after this PR severely restricts the versions of faraday, sorbet-runtime, jwt, and net-http-persistent, those releases are being excluded by Bundler in projects that have newer versions of those gems—most projects.

So, if it's true that this gem is completely broken with faraday 2.5, the new release didn't fix that: Projects that already use faraday 2.5 will never know that a new version of this gem has been released because Bundler will ignore it.

If the intent is to ensure that the "broken" releases of the gem (those that allow faraday >= 2.0), then the old releases of the gem would need to be yanked from rubygems. _This is an extreme measure and I don't recommend it_, but that would at least ensure that the broken releases of the gem are no longer used. It could be done in combination with a release of the gem that excludes the specific versions of faraday (or one of the other gems) that are known to be broken.

I'll try to follow up with a PR to address this.

**Comment by sealabcore** [comment]
Agree with @RobinDaugherty here, could this gem be fixed to work with 2.5 instead of pinning to such an old version (1.10)? Faraday 2.6 is out now and we can't upgrade our Stream library from 3.0 until this change gets reverted.

**Comment by sealabcore** [comment]
I didn't even know 3.1 was released because Bundler was ignoring it. And, Dependabot was ignoring this new version as well.

**Comment by ferhatelmas** [comment]
> could this gem be fixed to work with 2.5 instead of pinning to such an old version (1.10)

Absolutely. that should be the direction we should take. If you're preparing a PR, please let us know. Otherwise, we will look into it to relax deps and to make it working with new version. cc @ffenix113

**Comment by sealabcore** [comment]
> So, if it's true that this gem is completely broken with faraday 2.5, the new release didn't fix that: Projects that already use faraday 2.5 will never know that a new version of this gem has been released because Bundler will ignore it.

This is still an issue. Our app won't install stream-chat-ruby 3.2.0 because other gems we have installed require Faraday > 2.

Why did you downgrade all the way back down to version 1 of Faraday when it worked fine until 2.5 was released? Can you make stream-chat-ruby 3.2.1 which allows Faraday < 2.5 until you get a working gem with the latest version of Faraday? Or drop that dependency entirely?

We are stuck on an older version of this gem for now 😭

File: stream-chat.gemspec [hunk 1]
```diff
@@ -26,14 +26,14 @@ Gem::Specification.new do |gem|
...
-  gem.add_dependency 'faraday-multipart'
-  gem.add_dependency 'faraday-net_http_persistent'
-  gem.add_dependency 'jwt'
-  gem.add_dependency 'net-http-persistent'
-  gem.add_dependency 'sorbet-runtime'
-  gem.add_development_dependency 'rake'
-  gem.add_development_dependency 'rspec'
-  gem.add_development_dependency 'simplecov'
-  gem.add_development_dependency 'simplecov-console'
+  gem.add_dependency 'faraday', '= 1.10.0'
```
Comment (sealabcore) [inline stream-chat.gemspec:29]: This should be 

```ruby
gem.add_dependency 'faraday', '< 2.5'
```

**Pull request merged**